R/loadReg2.R

```diff
@@ -331,29 +333,46 @@ predictSolute.loadReg2 <- function(
     # errors on this point.
   }
   
+  #deal with mean water/calendar year options
+  #need to filter data down to compelete years, and then send 'total' option to rloadest
+  #data freq, leap years, whatever is in newdata ob
+  if(agg.by %in% c("mean water year", "mean calendar year")) {
+    newdata <- filterCompleteYears(data = newdata, time.step = load.model@fit$time.step, 
+                                   agg.by = agg.by)
+    agg.by = "total" #going to rloadest
+    if(nrow(newdata) > 176000) {
+      stop(paste(strwrap("Sorry, rloadest can't handle more than 176,000 data points, 
+                          and loadflex does not currently support aggregating uncertainties. 
+                          Please change the agg.by argument to a shorter period, or else 
+                         supply a shorter period of data for newdata"), collapse = "\n"))
+    }
+  }
+  
+  
   # Get around data size constraints of predLoad and predConc by splitting the 
   # prediction into smaller chunks as necessary. chunk.size is the maximum
   # number of values per chunk; rloadest can handle at most 176000 values by
   # default. nchunks is the number of chunks required.
-  chunk.size <- 176000
+  chunk.size <- 175872 #multiple of 96, so breaks iv data on day boundaries
   nchunks <- ceiling(nrow(newdata) / chunk.size)
   datachunks <- lapply(1:nchunks, function(i) {
     newdata[((i-1)*chunk.size + 1):min(i*chunk.size, nrow(newdata)),]
   })
-  
+  browser()
   # Now do the prediction for each chunk and then reassemble the full set of
   # predictions
   preds_lin_raw <- lapply(datachunks, function(datachunk) {
     switch(
       flux.or.conc,
       "flux"={
         # ... may contain seopt, print
-        predLoad(fit=load.model@fit, newdata=datachunk, by="unit", allow.incomplete=FALSE, conf.int=level, ...)
+        predLoad(fit=load.model@fit, newdata=datachunk, by=agg.by, allow.incomplete=FALSE, conf.int=level, ...)
       }, 
       "conc"={
         predLoad_args <- intersect(c('seopt','print'), names(list(...))) # load.units and conf.int have already been rejected above
         if(length(predLoad_args) > 0) warning(paste("these args are ignored for flux.or.conc='conc':", paste(predLoad_args, collapse=', ')))
-        predConc(fit=load.model@fit, newdata=datachunk, by="unit", allow.incomplete=FALSE, conf.int=level) 
+        #will fail here if agg.by != 'unit' - don't want to change aggregation behind user's back
```

Comment:
OK, so when you wrote "don't want to change aggregation behind user's back", are you simply justifying the change of the `by` argument from `"unit"` to `agg.by`? That part isn't super clear from your comment, especially because future code readers won't even be looking at this git diff.

If predConc does accept both `day` and `unit`, let's also update the comment to note that.